### PR TITLE
Implemented PhysicalObjectSystem to mimic the minio interface using a physical folder

### PR DESCRIPTION
**Pull request by HeDo88TH**
There are 48 tests that cover the majority of the implementation. It works well and fast but few edge cases are not tested and a couple of features are not supported yet. I plan to add more tests and features as soon as they are required or requested.

### CODE REVIEW

File: Registry.Common/Utils.cs [hunk 1]
```diff
@@ -24,5 +25,40 @@ public static string RandomString(int length)
...
+        {
+            return FilesAreEqual(new FileInfo(first), new FileInfo(second));
+        }
+
+        public static bool FilesAreEqual(FileInfo first, FileInfo second)
+        {
+            if (first.Length != second.Length)
+                return false;
+
+            if (string.Equals(first.FullName, second.FullName, StringComparison.OrdinalIgnoreCase))
```
Comment (pierotofy) [inline Registry.Common/Utils.cs:41]: I think this should be `StringComparison.Ordinal`. https://stackoverflow.com/a/9546069

```
No, filenames on S3 are case sensitive. 
```

"NaMe" != "name"
